osquery/tables/system/windows/system_info.cpp

```diff
@@ -36,12 +36,14 @@ QueryData genSystemInfo(QueryContext& context) {
   }
 
   WmiRequest wmiSystemReq("select * from Win32_ComputerSystem");
+  WmiRequest wmiSystemReqProc("select * from Win32_Processor");
   std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  std::vector<WmiResultItem>& wmiResultsProc = wmiSystemReqProc.results();
   if (wmiResults.size() != 0) {
     long numProcs = 0;
     wmiResults[0].GetLong("NumberOfLogicalProcessors", numProcs);
     r["cpu_logical_cores"] = INTEGER(numProcs);
-    wmiResults[0].GetLong("NumberOfProcessors", numProcs);
+    wmiResultsProc[0].GetLong("NumberOfCores", numProcs);
```

Comment:
Ensure you check that `wmiSystemReqProc.empty()` before using.

Reply:
Good call!